src/content/posts/4-way-handshake.md/index.md

```diff
@@ -0,0 +1,139 @@
+tags: ["network", "wifi"]
+author: clementS
+draft: false
+---
+
+## Abstract
+To connect to a WiFi network, the authentication depends on the WiFi protocol. The most common protocols currently in use are 
+1. **WEP**
+2. **WPA Personal**
+3. **WPA Entreprise**
+4. **WPA2-Personal**
+5. **WPA2-Enterprise**
+6. **WPA3**  
+
+> There are 2 different versions of WPA and WPA2 which work slightly differently in terms of authentication.
+
+While WEP and WPA3 have their own authentication protocols, WPA and WPA2 (which are the most common ones) use the **4-Way Handshake** method to allow a device to connect to WiFi and secure connection.
+
+The 4-Way Handshake requires a PMK (Pair Master Key), a secret shared by both sides (client and server).
+This can be a password (network key) or a username + password as it's the case for Enterprise authentication.
+
+In this article, you will learn how this method works from a network-oriented approach.
+
+## Configuration
+We will consider a WPA-Personal/WPA2-Personal WiFi, so the PMK will just be the WiFi password encrypted. We will go into further explains soon.  
+
+But remember that except getting PMK before authentication starts threw a WiFi password(secret shared by both sides), WPA2-Entreprise and WPA-Entreprise works the same way.
+Indeed, they use 4-way handshake to secure communication.  
+
+There is a device that knows the key (password) and wants to connect to a WPA/WPA2-PSK WiFi network.
+
+## 4-Way Handshake Initiation
+
+The device sends an authentication request to the network.
+**This request is not included in the 4-Way Handshake.**
+
+## 4-Way Handshake
+
+![schema](schema.png)
+
+This diagram shows the different requests and what they contain.
+
+> Note: *"STA" stands for Station, and "AP" for Access Point*  
+
+
+## First Request
+Firstly, the AP sends a nonce, which is a large random integer.
+
+When the station receives the packet, the station creates the PTK (Pairwise Transient Key) using the Key Derivation Function (KDF).
+This key is created with:
+1. The Pre-Shared Key (PSK),
+2. The ANonce received previously,
+3. The SNonce.
+
+The SNonce is similar to the ANonce, a large random number generated on the station side.
+Each time the station tries to authenticate, the SNonce and ANonce are generated again.
+These numbers ensure that the PTK is temporary.
+
+The KDF algorithm returns the PTK.
+We don’t use the password or the PSK directly because we don’t want the explicit key to go through the network.
+This is quite obvious; we don’t want others to easily intercept the password.
+
+### Pre-Shared Key (PSK)
+To get the PSK, the device only needs the SSID (network MAC address) and the password.  
+=> The process is really simple:  
+**Hash the password with the SSID as salt using the SHA-1 algorithm.**
+
+Now, we have the PSK (the AP also performs this operation).
+
+**The PSK is calculated before authentication starts (the station has all infos it needs).**  

```

Comment:
Besides the share secret , how did it get all this info ? it is actually very interesting to know what packets are broadcasted by the AP before any authentication, and what kind of information it contains. Idem for the stations that search to connect to known WiFi. It is not trivial. These packets can lead to multiple attacks, some research might be interesting !

Reply:
added clarity